pkg/controller/jobframework/integrationmanager_test.go

```diff
@@ -236,6 +249,92 @@ func compareCallbacks(x, y interface{}) bool {
 	return reflect.ValueOf(xcb.AddToScheme).Pointer() == reflect.ValueOf(ycb.AddToScheme).Pointer()
 }
 
+func TestRegisterExternal(t *testing.T) {
+	cases := map[string]struct {
+		manager         *integrationManager
+		integrationName string
+		jobType         runtime.Object
+		wantError       error
+		wantList        []string
+		wantJobType     runtime.Object
+	}{
+		"successful": {
+			manager: &integrationManager{
+				names: []string{"oldFramework"},
+				integrations: map[string]IntegrationCallbacks{
+					"oldFramework": testIntegrationCallbacks,
+				},
+			},
+			integrationName: "newFramework",
+			jobType:         &corev1.Pod{},
+			wantError:       nil,
+			wantList:        []string{"oldFramework"},
+			wantJobType:     &corev1.Pod{},
+		},
+		"duplicate name": {
+			manager: &integrationManager{
+				names: []string{"oldFramework"},
+				integrations: map[string]IntegrationCallbacks{
+					"oldFramework": testIntegrationCallbacks,
+				},
+				externalIntegrations: map[string]runtime.Object{
+					"newFramework": &corev1.Pod{},
+				},
+			},
+			integrationName: "newFramework",
+			jobType:         &corev1.Pod{},
+			wantError:       errDuplicateFrameworkName,
+			wantList:        []string{"oldFramework"},
+			wantJobType:     nil,
+		},
+		"duplicate with internal name": {
+			manager: &integrationManager{
+				names: []string{"newFramework"},
+				integrations: map[string]IntegrationCallbacks{
+					"newFramework": testIntegrationCallbacks,
+				},
+			},
+			integrationName: "newFramework",
+			jobType:         &corev1.Pod{},
+			wantError:       errDuplicateFrameworkName,
+			wantList:        []string{"newFramework"},
+			wantJobType:     &corev1.Pod{},
+		},
+		"missing JobType": {
+			manager:         &integrationManager{},
+			integrationName: "newFramework",
+			jobType:         nil,
+			wantError:       errMissingMandatoryField,
+			wantList:        []string{},
+		},
+	}
+
+	for tcName, tc := range cases {
+		t.Run(tcName, func(t *testing.T) {
+			gotError := tc.manager.registerExternal(tc.integrationName, tc.jobType)
+			if diff := cmp.Diff(tc.wantError, gotError, cmpopts.EquateErrors()); diff != "" {
+				t.Errorf("Unexpected error (-want +got):\n%s", diff)
+			}
+			gotList := tc.manager.getList()
+			if diff := cmp.Diff(tc.wantList, gotList); diff != "" {
+				t.Errorf("Unexpected frameworks list (-want +got):\n%s", diff)
+			}
+
+			if gotJobType, found := tc.manager.getExternal(tc.integrationName); found {
+				if diff := cmp.Diff(tc.jobType, gotJobType, cmp.FilterValues(func(_, _ interface{}) bool { return true }, cmp.Comparer(compareJobTypes))); diff != "" {
```

Comment:
what is the FilterValues for?

Reply:
reworked to eliminate need for a custom compare function